server/src/yml-objects/AbstractYmlObject.ts

```diff
@@ -75,37 +84,71 @@ export abstract class AbstractYmlObject implements CompletionItem {
      * that are out of the current scope.
      */
     public enrichWith(
-        fields: FieldContext[],
-        connection: IConnection,
+        documentation: string,
+        type: string,
         sourceElementName?: string,
-        baseType?: string,
         scopeStartOffset?: number,
         scopeEndOffset?: number,
     ): void {
-        sourceElementName = sourceElementName ? sourceElementName : 'static';
+        this.sourceElementName = sourceElementName ?? 'static';
         this.data = `id_${sourceElementName}_${this.label}`;
-        this.setDocumentation(getDocumentation(fields, connection));
-        this.detail = getType(fields, connection, baseType);
+        this.setUserInformations(this.buildDetailString(type), documentation);
         if (scopeEndOffset && scopeStartOffset) {
             this.scopeEndOffset = scopeEndOffset;
             this.scopeStartOffset = scopeStartOffset;
         }
     }
 
+    protected buildDetailString(type: string): string {
+        const separator = this.sourceElementName === 'static' ? ' ' : '.';
+        return `(${this.kindName}) [${this.sourceElementName}]${separator}${this.label} ⇒ ${type}`;
+    }
+
+    public getHoverContent(): MarkupContent {
+        return this.hoverContent;
+    }
+
+    public hasDocumentation(): boolean {
+        return !!this.documentation || !!this.detail;
+    }
+
     /**
-     * Set the documentation of this object as a `MarkupContent`.
-     * Set the documentation to `null` if empty.
+     * Set the documentation of this object as a `MarkupContent` its detail attribute
+     * as well as its representation when hovering it.
+     * The documentation should always exist and consists at least as the value of detail.
+     *
+     * @param details Some details about this object, as a Markdown string.
      * @param doc The documentation of this object, as a Markdown string.
      *
      * @see MarkupContent
      */
-    public setDocumentation(doc: string): void {
-        if (!doc) {
-            this.documentation = null;
+    public setUserInformations(details: string, doc: string): void {
+        if (!details && !doc) {
+            return;
         }
-        this.documentation = {
+
+        // If possible, set documentation's value.
+        if (!!doc) {
+            this.documentation = {
+                kind: MarkupKind.Markdown,
+                value: doc,
+            };
+        }
+
+        // There is no value for `details`. This means that we have a documentation.
```

Comment:
The implication “no details ⇒ we have a documentation” is obscure and rather counter-intuitive to outsiders like me.

Reply:
OK. Is it better?
```suggestion
        // There is no value for `details`. This means that we have a value for `doc`.
```

Reply:
Let me remind you that we have this code a few lines above:
```ts
if (!details && !doc) {
    return;
}
```

Reply:
AH LOL I was thinking at a higher level of abstraction, like “there is some weird inner logic everywhere in the extension that ensures that when foo then bar”.

Reply:
OK for your sugg